src/acreplace.rs

```diff
@@ -1,22 +1,64 @@
-use aho_corasick::AhoCorasickBuilder;
-use aho_corasick::AhoCorasick;
+use aho_corasick::{AhoCorasick, AhoCorasickBuilder, MatchKind};
 use std::{
     cell::RefCell,
     collections::hash_map::HashMap
 };
+use serde::Deserialize;
 
 struct Replacements {
     pub automaton: AhoCorasick,
-    pub replacements: Vec<String>
+    pub replacements: Vec<String>,
+}
+
+#[derive(Deserialize)]
+struct AhoCorasickOptions {
+    #[serde(default, deserialize_with = "deserialize_byond_bool")]
+    pub anchored: bool,
+    #[serde(default, deserialize_with = "deserialize_byond_bool")]
+    pub ascii_case_insensitive: bool,
+    #[serde(default, deserialize_with = "deserialize_matchkind")]
+    pub match_kind: MatchKind,
+}
+
+impl AhoCorasickOptions {
+    fn auto_configure_and_build(&self, patterns: &Vec<String>) -> AhoCorasick {
+        AhoCorasickBuilder::new()
+        .anchored(self.anchored)
+        .ascii_case_insensitive(self.ascii_case_insensitive)
+        .match_kind(self.match_kind)
+        .auto_configure(patterns)
+        .build(patterns)
+    }
+}
+
+fn deserialize_byond_bool<'de, D>(deserializer: D) -> Result<bool, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
+    match u8::deserialize(deserializer)? {
+        1 => Ok(true),
```

Comment:
Shouldn't this be the other way around? Any non-zero value is true?

Reply:
The default bool value is false so I thought it fitting, but since it's definitely a number at this point it would make more sense indeed